src/main/java/com/coreos/jetcd/exception/ErrorCode.java

```diff
@@ -0,0 +1,66 @@
+package com.coreos.jetcd.exception;
+
+import io.grpc.Status;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public enum ErrorCode {
+
+  CANCELLED(Status.CANCELLED),
+  UNKNOWN(Status.UNKNOWN),
+  INVALID_ARGUMENT(Status.INVALID_ARGUMENT),
+  DEADLINE_EXCEEDED(Status.DEADLINE_EXCEEDED),
+  NOT_FOUND(Status.NOT_FOUND),
+  ALREADY_EXISTS(Status.ALREADY_EXISTS),
+  PERMISSION_DENIED(Status.PERMISSION_DENIED),
+  UNAUTHENTICATED(Status.UNAUTHENTICATED),
+  RESOURCE_EXHAUSTED(Status.RESOURCE_EXHAUSTED),
+  FAILED_PRECONDITION(Status.FAILED_PRECONDITION),
+  ABORTED(Status.ABORTED),
+  OUT_OF_RANGE(Status.OUT_OF_RANGE),
+  UNIMPLEMENTED(Status.UNIMPLEMENTED),
+  INTERNAL(Status.INTERNAL),
+  UNAVAILABLE(Status.UNAVAILABLE),
+  DATA_LOSS(Status.DATA_LOSS),;
+
+  private static final Map<Integer, ErrorCode> errorByRpcCode;
+
+  static {
+    Map<Integer, ErrorCode> realMap = new LinkedHashMap<>();
+    for (ErrorCode errorCode : ErrorCode.values()) {
+      realMap.put(errorCode.getCode(), errorCode);
+    }
+    errorByRpcCode = Collections.unmodifiableMap(realMap);
+  }
+
+  private final Status.Code code;
+
+  ErrorCode(Status status) {
+    this.code = status.getCode();
+  }
+
+  int getCode() {
+    return this.code.value();
+  }
+
+  /**
+   * Returns the error code represents by {@code name}, or {@code defaultValue} if {@code name} does
+   * not map to a known code.
+   */
+  static ErrorCode valueOf(String name, ErrorCode defaultValue) {
+    try {
+      return ErrorCode.valueOf(name);
+    } catch (IllegalArgumentException e) {
+      return defaultValue;
+    }
+  }
+
+  /**
+   * Returns the error code corresponding to a gRPC status, or {@code UNKNOWN} if not recognized.
+   */
+  static ErrorCode fromGrpcStatus(Status status) {
+    ErrorCode code = errorByRpcCode.get(status.getCode().value());
```

Comment:
Not sure if we need ``errorByRpcCode`` as we can loop through the values, similar to what is done in the static initialization block.